### PR TITLE
Fix Balsamic WGS runs

### LINKED ISSUE #566: Unable to start BALSAMIC-wgs analysis
A production team member should be able to start balsamic-wgs analysis.

Problem: 
In Balsamic, this is done by not providing a `--target-bed` file, which toggles WGS mode. As of now, it seems it is mandatory to provide a `--target-bed` file: https://github.com/Clinical-Genomics/cg/blob/master/cg/cli/workflow/balsamic/base.py#L231

Steps to reproduce
  1. up or us
  2. cg workflow balsamic -c eagerserval -e han.velsing@alucard.nomnom

```
Traceback (most recent call last):
  File "/home/proj/bin/conda/envs/P_main/bin/cg", line 11, in <module>
    sys.exit(base())
  File "/home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages/click/core.py", line 1043, in invoke
    return Command.invoke(self, ctx)
  File "/home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages/cg/cli/workflow/balsamic/base.py", line 75, in balsamic
    context.invoke(config_case, case_id=case_id, target_bed=target_bed)
  File "/home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages/cg/cli/workflow/balsamic/base.py", line 239, in config_case
    raise BalsamicStartError("No target bed specified!")
cg.exc.BalsamicStartError
```

Expected outcome / suggested solution:
A coffee in my cup.

Questions: 
Q: Does the same problem occur for tea?
A: Yes.

What needs to be done on the planning meeting:
- Estimate storypoints
- **Remember to add label *PBL Teamname***

See https://github.com/Clinical-Genomics/development/blob/master/git/issue-reports.md for more!

### PR DESCRIPTION
**Pull request by hassanfa**
This PR adds/fixes #566 

**How to prepare for test**:
- [x] ssh to hasta
- [x] install on stage:
`bash servers/resources/hasta.sclifelab.se/update-cg-stage.sh fix_wgs_balsamic`
- [x] change to stage: `us`

In order to test, we need to alter lims in cg-stage.yaml to production. Do the following:
- [x] change lims-stage to lims:
```
sed "s|clinical-lims-stage|clinical-lims|g" \
/home/proj/stage/servers/config/hasta.scilifelab.se/cg-stage.yaml \
> ~/cg-stage.fix_wgs_balsamic.yaml
```

**How to test**:
Link, create config, and run analysis:
- [x] `cg --config ~/cg-stage.fix_wgs_balsamic.yaml workflow balsamic -c fleetjay`

**Expected test outcome**:
- [x] fastq files should link to `/home/proj/stage/cancer/cases/fleetjay/fastq`
- [x] a config file should be created in: `/home/proj/stage/cancer/cases/fleetjay/fleetjay.config`
- [x] Analysis jobs should be submitted to hasta

**Review:**
- [x] code approved by @patrikgrenfeldt 
- [x] tests executed by @hassanfa 
- [x] "Merge and deploy" approved by @patrikgrenfeldt 
Thanks for filling in who performed the code review and the test!

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

**Comment by hassanfa** [comment]
_Placeholder for tests for a TGA sample_

**Comment by hassanfa** [comment]
A TGA sample: grandpony

test for drying run being correct:
image

Tests for linking:
image

Tests for correctly picking up application, creating config, and running the workflow:
image

Tests for jobs properly submitted and running:
image

**Comment by hassanfa** [comment]
One test and blacking problem... Looking into it...

**Comment by hassanfa** [comment]
Travis fails on building from archive, but rest of the tests pass. Ready for review!

**Comment by hassanfa** [comment]
New pylint complains on too many input arguments is weird. How can I turn it off or make it not dislike my code? :-)  e.g. tests/cli/workflow/balsamic/conftest.py:244

File: tests/cli/workflow/balsamic/test_cli_balsamic_config_case.py [hunk 1]
```diff
@@ -128,6 +128,25 @@ def test_target_bed_from_lims(
...
+    # GIVEN case with wgs tag
+    bed_key = "-p"
+    case_id = balsamic_case_wgs.internal_id
+
+    # WHEN dry running
+    result = cli_runner.invoke(
+        config_case, [case_id, "--dry-run"], obj=balsamic_context
+    )
+
+    # THEN dry-print should NOT include the bed_key and the bed_value including path
```
Comment (patrikgrenfeldt) [inline tests/cli/workflow/balsamic/test_cli_balsamic_config_case.py]:
```suggestion
    # THEN dry-print should NOT include the bed_key
```

File: tests/cli/workflow/balsamic/test_cli_balsamic_config_case.py [hunk 2]
```diff
@@ -128,6 +128,25 @@ def test_target_bed_from_lims(
     assert str(bed_path) in result.output
 
 
+def test_wgs_from_lims(
```
Comment (patrikgrenfeldt) [inline tests/cli/workflow/balsamic/test_cli_balsamic_config_case.py]:
```suggestion
def test_wgs_excludes_bed_for_balsamic(
```

File: cg/cli/workflow/balsamic/base.py [hunk 3]
```diff
@@ -143,7 +143,19 @@ def config_case(
     conda_env = context.obj["balsamic"]["conda_env"]
     root_dir = context.obj["balsamic"]["root"]
     wrk_dir = Path(f"{root_dir}/{case_id}/fastq")
+    application_type = set()
```
Comment (patrikgrenfeldt) [inline cg/cli/workflow/balsamic/base.py]:
```suggestion
    application_types = set()
```

File: cg/cli/workflow/balsamic/base.py [hunk 4]
```diff
@@ -143,7 +143,19 @@ def config_case(
...
+    acceptable_application = {"wgs", "wes", "tgs"}
```
Comment (patrikgrenfeldt) [inline cg/cli/workflow/balsamic/base.py]:
```suggestion
    acceptable_applications = {"wgs", "wes", "tgs"}
```

File: cg/cli/workflow/balsamic/base.py [hunk 5]
```diff
@@ -232,7 +255,7 @@ def config_case(
     elif nr_normal_paths > 1:
         raise BalsamicStartError("Too many normal samples found: %s" % nr_normal_paths)
 
-    if not target_bed:
+    if not target_bed and application_type.issubset({"wes", "tgs"}):
```
Comment (patrikgrenfeldt) [inline cg/cli/workflow/balsamic/base.py]:
```suggestion
    if not target_bed and application_type.issubset(applications_requiring_bed):
```

Comment (patrikgrenfeldt) [inline cg/cli/workflow/balsamic/base.py]: raise if wgs and --target was given through cli

**Comment by keyvanelhami** [comment]
@patrikgrenfeldt Please update me and @emmser when this issue is deployed.

**Comment by hassanfa** [comment]
@patrikgrenfeldt pylint and tests and black passed. Awaiting to merge/deploy this. Peace out! ✋

**Comment by patrikgrenfeldt** [comment]
deployed on hasta and clinical-db @keyvanelhami @emmser

**Pull request merged**